packages/admin-next/dashboard/src/routes/draft-orders/draft-order-create/components/create-draft-order-form/create-draft-order-drawer/add-custom-item-drawer.tsx

```diff
@@ -0,0 +1,129 @@
+import { Button, CurrencyInput, Hint, Input, Label } from "@medusajs/ui"
+import { useState } from "react"
+import { useTranslation } from "react-i18next"
+import { z } from "zod"
+import { SplitView } from "../../../../../../components/layout/split-view"
+import { CustomItemSchema } from "../constants"
+import { useCreateDraftOrder } from "../hooks"
+import { CustomItem } from "../types"
+
+export const AddCustomItemDrawer = () => {
+  const { region, custom } = useCreateDraftOrder()
+  const { currency } = region || {}
+
+  const currencyCode = currency?.code || ""
+  const nativeSymbol = currency?.symbol_native || ""
+
+  const { t } = useTranslation()
+
+  const [item, setItem] = useState<Partial<CustomItem>>({
+    title: "",
+    quantity: 1,
+    unit_price: undefined,
+  })
+  const [errors, setErrors] = useState<z.ZodError<
+    z.infer<typeof CustomItemSchema>
+  > | null>(null)
+
+  const handleSave = () => {
+    const parsed = CustomItemSchema.safeParse(item)
+
+    if (!parsed.success) {
+      setErrors(parsed.error)
+
+      return
+    }
+
+    custom.update(parsed.data)
+  }
+
+  return (
+    <div className="flex size-full flex-col overflow-hidden">
+      <div className="size-full flex-1 overflow-auto px-6 py-4">
+        <div className="flex flex-col gap-y-4 [&>div]:flex [&>div]:flex-col [&>div]:gap-y-2">
+          <div>
+            <Label weight="plus" size="small">
+              {t("fields.title")}
+            </Label>
+            <Input
+              value={item.title}
+              onChange={(e) =>
+                setItem((prev) => ({ ...prev, title: e.target.value }))
+              }
+            />
+            <ErrorMessage errors={errors} field="title" />
+          </div>
+          <div>
+            <Label weight="plus" size="small">
+              {t("fields.quantity")}
+            </Label>
+            <Input
+              value={item.quantity}
+              type="number"
+              step={1}
+              onChange={(e) => {
+                const number = Number(e.target.value)
+                const value = isNaN(number) ? undefined : number
+
+                setItem((prev) => ({ ...prev, quantity: value }))
+              }}
+            />
+            <ErrorMessage errors={errors} field="quantity" />
+          </div>
+          <div>
+            <Label weight="plus" size="small">
+              {t("fields.unitPrice")}
+            </Label>
+            <CurrencyInput
```

Comment:
todo: something is off with this component, I cannot enter decimal point/comma when number is smaller than 1000. Also sometimes when a number is entered I cannot add the item because the validation error says price is required.

Reply:
Works now, but when I input a comma or enter two dots accidentally I get `NaN`